### ISSUE TITLE
The link prediction code via GCN cannot work properly

### ISSUE THREAD

**Issue by ZihanChen1995** [stellargraph/stellargraph#1786]
### Describe the bug

Hi! I'm trying to work with the [link prediction code via GCN](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/gcn-link-prediction.html). However, I found the acc doesn't change through both the training and testing process. I firstly reproduced the code locally, it cannot work as the document describes. Hence I moved to the online Jupyter Notebook on Colab, but the problem still existed. Do you have any idea why this happens? Thank you a lot!

### To Reproduce

I strictly followed the steps one-by-one. [gcn-link-prediction.ipynb](https://colab.research.google.com/github/stellargraph/stellargraph/blob/master/demos/link-prediction/gcn-link-prediction.ipynb)

### Observed behavior/Expected behavior

The acc before training is supposed to be 0.50, and should be improved through training. However, it keeps 0.00 all the time.

```
print("\nTrain Set Metrics of the initial (untrained) model:")
for name, val in zip(model.metrics_names, init_train_metrics):
    print("\t{}: {:0.4f}".format(name, val))

print("\nTest Set Metrics of the initial (untrained) model:")
for name, val in zip(model.metrics_names, init_test_metrics):
    print("\t{}: {:0.4f}".format(name, val))

1/1 [==============================] - 0s 2ms/step - loss: 1.7449 - acc: 0.0000e+00
1/1 [==============================] - 0s 2ms/step - loss: 1.7606 - acc: 0.0000e+00

Train Set Metrics of the initial (untrained) model:
	loss: 1.7449
	acc: 0.0000

Test Set Metrics of the initial (untrained) model:
	loss: 1.7606
	acc: 0.0000
```

After training:
```
1/1 [==============================] - 0s 2ms/step - loss: 0.2774 - acc: 0.0000e+00
1/1 [==============================] - 0s 1ms/step - loss: 0.7085 - acc: 0.0000e+00

Train Set Metrics of the trained model:
	loss: 0.2774
	acc: 0.0000

Test Set Metrics of the trained model:
	loss: 0.7085
	acc: 0.0000
```

The loss is decreasing, but the acc has no change:
```
1/1 - 0s - loss: 0.4074 - acc: 0.0000e+00 - val_loss: 0.6118 - val_acc: 0.0000e+00
Epoch 42/50
1/1 - 0s - loss: 0.3902 - acc: 0.0000e+00 - val_loss: 0.6148 - val_acc: 0.0000e+00
Epoch 43/50
1/1 - 0s - loss: 0.4015 - acc: 0.0000e+00 - val_loss: 0.6307 - val_acc: 0.0000e+00
Epoch 44/50
1/1 - 0s - loss: 0.3842 - acc: 0.0000e+00 - val_loss: 0.6386 - val_acc: 0.0000e+00
Epoch 45/50
1/1 - 0s - loss: 0.3881 - acc: 0.0000e+00 - val_loss: 0.6706 - val_acc: 0.0000e+00
Epoch 46/50
1/1 - 0s - loss: 0.3536 - acc: 0.0000e+00 - val_loss: 0.6956 - val_acc: 0.0000e+00
Epoch 47/50
1/1 - 0s - loss: 0.4018 - acc: 0.0000e+00 - val_loss: 0.7163 - val_acc: 0.0000e+00
Epoch 48/50
1/1 - 0s - loss: 0.3828 - acc: 0.0000e+00 - val_loss: 0.7294 - val_acc: 0.0000e+00
Epoch 49/50
1/1 - 0s - loss: 0.3993 - acc: 0.0000e+00 - val_loss: 0.7215 - val_acc: 0.0000e+00
Epoch 50/50
1/1 - 0s - loss: 0.3623 - acc: 0.0000e+00 - val_loss: 0.7085 - val_acc: 0.0000e+00

```

### Environment

Operating system: Both Ubuntu 18.04 and Google Colab

Package versions: stellargraph==stable version

### Addition Details

I guess the problem happens in this step:`init_train_metrics = model.evaluate(train_flow)`. The model is supposed to output the same array shape like ground truth label (e.g. `array[0,1,0...]`). However, the output becomes a 1x2 array:
```
print(init_train_metrics)

[1.7448515892028809, 0.0]
```

Thank you in advance!

**Comment by ZihanChen1995** [comment]
I just found this is a duplicate question with [issue #1766](https://github.com/stellargraph/stellargraph/issues/1766). And the problem is caused by Tensorflow version, since they change `metric` dramatically. Here is a reference solution from issue 1766:

1. replace acc with binary_accuracy: model.compile(..., metrics=["binary_accuracy"]) (which is what #1770 does in the demo in the repo)
2. remove the Reshapeing
3. use TF 2.1 with stellargraph 1.2.1 (that is, version 2.1 seems to work for me, not just 2.0), where fit, evaluate and predict should work too (no need to use ..._generator)
